internal/provider/resource_saml_group_mapping.go

```diff
@@ -0,0 +1,380 @@
+            _stub
+          }
+	}`
+
+	samlIdpID := d.Get("saml_idp_id").(string)
+	providerGroupID := d.Get("provider_group_id").(string)
+	role := d.Get("role").(string)
+	projects := utils.ConvertListToString(d.Get("projects").([]interface{}))
+
+	// populate the graphql variables
+	vars := &wiz.UpdateSAMLGroupMappingInput{}
+	vars.ID = samlIdpID
+	vars.Patch.Upsert.ProviderGroupID = providerGroupID
+	vars.Patch.Upsert.Role = role
+	vars.Patch.Upsert.Projects = projects
+
+	// process the request
+	data := &wiz.UpdateSAMLGroupMappingPayload{}
+	requestDiags := client.ProcessRequest(ctx, m, vars, data, query, "saml_group_mapping", "update")
+	diags = append(diags, requestDiags...)
+	if len(diags) > 0 {
+		return diags
+	}
+
+	return resourceSAMLGroupMappingRead(ctx, d, m)
+}
+
+func resourceSAMLGroupMappingDelete(ctx context.Context, d *schema.ResourceData, m interface{}) (diags diag.Diagnostics) {
+	tflog.Info(ctx, "resourceWizSAMLGroupMappingDelete called...")
+
+	// check the id
+	if d.Id() == "" {
+		return nil
+	}
+
+	// define the graphql query
+	query := `mutation SetSAMLGroupMapping ($input: ModifySAMLGroupMappingInput!) {
+	  modifySAMLIdentityProviderGroupMappings(input: $input) {
+            _stub
+          }
+	}`
+
+	samlIdpID := d.Get("saml_idp_id").(string)
+	providerGroupID := d.Get("provider_group_id").(string)
+
+	// populate the graphql variables
+	vars := &wiz.DeleteSAMLGroupMappingInput{}
+	vars.ID = samlIdpID
+	vars.Patch.Delete = []string{providerGroupID}
+
+	// process the request
+	data := &wiz.UpdateSAMLGroupMappingPayload{}
+	requestDiags := client.ProcessRequest(ctx, m, vars, data, query, "saml_group_mapping", "delete")
+	diags = append(diags, requestDiags...)
+	if len(diags) > 0 {
+		return diags
+	}
+
+	return diags
+}
+
+func querySAMLGroupMappings(ctx context.Context, m interface{}, samlIdpID string, providerGroupID string, roleId string, projectIDs []string) (*wiz.SAMLGroupMapping, diag.Diagnostics) {
+	// define the graphql query
+	query := `query samlIdentityProviderGroupMappings ($id: ID!, $first: Int! $after: String){
+	    samlIdentityProviderGroupMappings (
+	        id: 	$id,
+			first: 	$first
+			after: 	$after
+	    ) {
+			pageInfo {
+				  hasNextPage
+				  endCursor
+			}
+	        nodes {
+			  providerGroupId
+			  role {
+				description
+				id
+				isProjectScoped
+				name
+				scopes
+			  }
+			  projects {
+				id
+			  }
+			}
+	    }
+	}`
+
+	// populate the graphql variables
+	vars := &internal.QueryVariables{}
+	vars.ID = samlIdpID
+	vars.First = 100
+
+	var matchingNode *wiz.SAMLGroupMapping
+	// Since we can't filter by providerGroupId server side we have to do it client side
+	// Execute the query in a loop until we found the group we are looking for, or all pages have been fetched
+
+	found := false
+	for !found {
+		data := &ReadSAMLGroupMappings{}
+		requestDiags := client.ProcessRequest(ctx, m, vars, data, query, "saml_idp", "read")
```

Comment:
Please take a look at `ProcessPagedRequest` in the client package and data sources that implement, the idea is to abstract these complexities out of the provider resources